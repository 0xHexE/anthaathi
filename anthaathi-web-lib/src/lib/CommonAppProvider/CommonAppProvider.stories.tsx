import { ComponentMeta } from '@storybook/react';

import { CommonAppProvider } from './CommonAppProvider';

export default {
  component: CommonAppProvider,
} as ComponentMeta<typeof CommonAppProvider>;
