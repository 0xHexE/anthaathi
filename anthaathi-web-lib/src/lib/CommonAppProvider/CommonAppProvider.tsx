import React from 'react';
import { BaseProvider, LightTheme } from 'baseui';
import { Provider as StyletronProvider } from 'styletron-react';
import { Client as Styletron } from 'styletron-engine-atomic';

const engine = new Styletron();

export interface CommonAppProviderProps {
  children: React.ReactNode;
}

export function CommonAppProvider({ children }: CommonAppProviderProps) {
  return (
    <StyletronProvider value={engine}>
      <BaseProvider theme={LightTheme}>{children}</BaseProvider>
    </StyletronProvider>
  );
}
