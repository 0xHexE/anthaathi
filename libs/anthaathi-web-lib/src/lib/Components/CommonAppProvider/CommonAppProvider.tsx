import * as React from 'react';
import { BaseProvider, createLightTheme } from 'baseui';
import { Provider as StyletronProvider } from 'styletron-react';
import { Client, Server } from 'styletron-engine-atomic';
import { Block } from 'baseui/block';
import type { Theme } from 'baseui';

const getHydrateClass = () =>
  document.getElementsByClassName(
    '_styletron_hydrate_',
  ) as HTMLCollectionOf<HTMLStyleElement>;

export const engine =
  typeof window === 'undefined'
    ? new Server()
    : new Client({
        hydrate: getHydrateClass(),
      });

const overrides: Partial<Theme> = {
  typography: {
    HeadingLarge: {
      fontFamily: "'Red Hat Display', sans-serif",
    },
    HeadingMedium: {
      fontFamily: "'Red Hat Display', sans-serif",
    },
    HeadingSmall: {
      fontFamily: "'Red Hat Display', sans-serif",
    },
    HeadingXSmall: {
      fontFamily: "'Red Hat Display', sans-serif",
    },
    HeadingXLarge: {
      fontFamily: "'Red Hat Display', sans-serif",
    },
    HeadingXXLarge: {
      fontFamily: "'Red Hat Display', sans-serif",
    },
  } as Theme['typography'],
};

const LightTheme = createLightTheme(
  {
    primaryFontFamily: "'IBM Plex Sans', sans-serif",
  },
  {
    borders: {
      useRoundedCorners: true,
      buttonBorderRadius: '8px',
      inputBorderRadius: '8px',
      popoverBorderRadius: '8px',
      surfaceBorderRadius: '8px',
    },
    ...overrides,
  },
);

export interface CommonAppProviderProps {
  children: React.ReactNode;
}

export function CommonAppProvider({ children }: CommonAppProviderProps) {
  return (
    <StyletronProvider value={engine}>
      <BaseProvider theme={LightTheme}>
        <Block
          minHeight="100vh"
          backgroundColor="backgroundSecondary"
          display="block"
        >
          {children}
        </Block>
      </BaseProvider>
    </StyletronProvider>
  );
}
