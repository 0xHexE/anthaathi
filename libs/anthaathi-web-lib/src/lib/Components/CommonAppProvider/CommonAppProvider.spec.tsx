import * as React from 'react';
import { render } from '@testing-library/react';
import { CommonAppProvider } from './CommonAppProvider';

describe('CommonAppProvider', () => {
  it('should render', () => {
    const element = render(<CommonAppProvider>test</CommonAppProvider>);
    expect(element.container).toMatchSnapshot();
  });
});
