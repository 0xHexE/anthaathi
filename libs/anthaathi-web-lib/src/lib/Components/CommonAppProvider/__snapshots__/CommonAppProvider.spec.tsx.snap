// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommonAppProvider should render 1`] = `
<div>
  <div
    class=""
  >
    <div
      class="_ae _af _ag"
      data-baseweb="block"
    >
      test
    </div>
  </div>
  <div
    class=""
  />
</div>
`;
